sympy/physics/units/dimensions.py

```diff
@@ -437,18 +438,32 @@ def _get_dimensional_dependencies_for_name(self, name):
             raise TypeError("Only equivalent dimensions can be added or subtracted.")
 
         if name.is_Pow:
-            dim = get_for_name(name.base)
-            return {k: v*name.exp for (k, v) in dim.items()}
+            dim_base = get_for_name(name.base)
+            dim_exp = get_for_name(name.exp)
+            if dim_exp != {}:
+                raise TypeError("The exponent for the power operator must be dimensionless.")
+            return {k: v*name.exp for (k, v) in dim_base.items()}
 
         if name.is_Function:
             args = (Dimension._from_dimensional_dependencies(
                 get_for_name(arg)) for arg in name.args)
             result = name.func(*args)
 
+            dicts = [get_for_name(i) for i in name.args]
+
             if isinstance(result, Dimension):
                 return self.get_dimensional_dependencies(result)
             elif result.func == name.func:
-                return {}
+                if isinstance(name, (sin, cos, tan, cot, sec, csc)):
```

Comment:
Well, there is an even better idea actually. There is an abstract class `TrigonometricFunction` (in `sympy.functions.elementary.trigonometric`), all these classes are subclasses of `TrigonometricFunction`. Maybe simply better `isinstance(name, TrigonometricFunction)`.

Reply:
(sorry for not telling before about that)

Reply:
Yes, good point, that makes it simpler. I've made the change.